finality-aleph/src/aggregation/aggregator.rs

```diff
@@ -178,19 +178,19 @@ impl<
         }
     }
 
-    pub(crate) async fn next_multisigned_hash(&mut self) -> Option<(H, PMS)> {
+    pub(crate) async fn next_multisigned_hash(&mut self) -> Result<Option<(H, PMS)>, ()> {
         loop {
             trace!(target: "aleph-aggregator", "Entering next_multisigned_hash loop.");
             match self.aggregator.try_pop_hash() {
                 Ok(res) => {
-                    return Some(res);
+                    return Ok(Some(res));
                 }
                 Err(AggregatorError::LastHashPlaced) => {
                     debug!(
                         target: "aleph-aggregator",
                         "Terminating next_multisigned_hash because the last hash has been signed."
                     );
-                    return None;
+                    return Ok(None);
```

Comment:
I would say it's better to just remove this case – we already ensure in other places in the code that we will not attempt to create silly justifications here, they would not be accepted anyway, and this is the only (I believe?) component that checks its own special termination criterion.

If that whole refactor adding `Result`s to spawning was there only to support this, but you think it's actually a good idea in general you can leave it in place.

Reply:
Yep, this `Result<>` was added here to support a scenario where a task finishes early but without errors.

Reply:
This might be good in general, but I would remove this specific case – this component really shouldn't be ending prematurely, just because it thinks that it's not responsible for any more blocks.

Reply:
I am not sure what to do with this one. If I am supposed to remove this behavior, then probably the whole `AggregatorError::LastHashPlaced` should be removed, together with its `notify_last_hash` method - this component shouldn't know about any termination condition, caller should just stop querying it.

Reply:
Yup, I would say that's exactly what should happen.